x/auth/ante/sigverify_test.go

```diff
@@ -72,6 +72,47 @@ func TestSetPubKey(t *testing.T) {
 	}
 }
 
+// TestSetPubKey_UnorderedNoEvents tests that when the tx is unordered, the sequence event is not emitted.
+func TestSetPubKey_UnorderedNoEvents(t *testing.T) {
+	suite := SetupTestSuite(t, true)
+	suite.txBuilder = suite.clientCtx.TxConfig.NewTxBuilder()
+
+	// keys and addresses
+	priv1, _, addr1 := testdata.KeyTestPubAddr()
+	priv2, _, addr2 := testdata.KeyTestPubAddr()
+	priv3, _, addr3 := testdata.KeyTestPubAddrSecp256R1(t)
+
+	addrs := []sdk.AccAddress{addr1, addr2, addr3}
+
+	msgs := make([]sdk.Msg, len(addrs))
+	// set accounts and create msg for each address
+	for i, addr := range addrs {
+		acc := suite.accountKeeper.NewAccountWithAddress(suite.ctx, addr)
+		require.NoError(t, acc.SetAccountNumber(uint64(i+1000)))
+		suite.accountKeeper.SetAccount(suite.ctx, acc)
+		msgs[i] = testdata.NewTestMsg(addr)
+	}
+	require.NoError(t, suite.txBuilder.SetMsgs(msgs...))
+	suite.txBuilder.SetFeeAmount(testdata.NewTestFeeAmount())
+	suite.txBuilder.SetGasLimit(testdata.NewTestGasLimit())
+
+	privs, accNums, accSeqs := []cryptotypes.PrivKey{priv1, priv2, priv3}, []uint64{0, 0, 0}, []uint64{0, 0, 0}
```

Comment:
Seems like a lot of test boilerplate just to get here. Is this copied from another test? Can we just pull out a reusable func?

Reply:
i reduced the boilerplate to strictly whats needed. fwiw, multiple tests seem to copy some variation of this, and id prefer to kick the can of generalizing this down the road

Reply:
[Reduce boilerplate](https://github.com/cosmos/cosmos-sdk/pull/24581/commits/6b7889bd497e5251683fbcf32bece41eaceba1b6)